app/uk/gov/hmrc/cgtpropertydisposals/service/returns/ReturnsService.scala

```diff
@@ -113,6 +123,79 @@ class DefaultReturnsService @Inject() (
     } yield returnResponse
   }
 
+  private def deltaCharge(
+    response: DesSubmitReturnResponse,
+    cgtReference: CgtReference,
+    fromDate: LocalDate,
+    toDate: LocalDate
+  )(implicit hc: HeaderCarrier): EitherT[Future, Error, Option[DeltaCharge]] =
+    response.ppdReturnResponseDetails.chargeReference match {
+      case Some(chargeReference) =>
+        lazy val charge = for {
```

Comment:
you don't need the `laazy val` here, you can just have the for-comprehension by itself